test/vmtests/vmtests/utils/libvirt_utils.py

```diff
@@ -30,14 +125,25 @@ def create_libvirt_domain_xml(vm_spec: VmSpec) -> str:
     vcpu.text = str(vm_spec.core_count)
 
     os_tag = ET.SubElement(domain, "os")
-    os_tag.attrib["firmware"] = "efi"
+    if boot_type == "efi" and host_os == "Ubuntu":
+        os_tag.attrib["firmware"] = "efi"
 
     os_type = ET.SubElement(os_tag, "type")
     os_type.text = "hvm"
 
-    firmware = ET.SubElement(domain, "firmware")
-    firmware.attrib["secure-boot"] = "yes"
-    firmware.attrib["enrolled-keys"] = "yes"
+    if boot_type == "efi" and host_os == "Ubuntu":
+        firmware = ET.SubElement(domain, "firmware")
+        firmware.attrib["secure-boot"] = "yes"
+        firmware.attrib["enrolled-keys"] = "yes"
+
+    nvram = ET.SubElement(os_tag, "nvram")
+
+    if boot_type == "efi":
+        loader = ET.SubElement(os_tag, "loader")
+        loader.attrib["readonly"] = "yes"
+        loader.attrib["secure"] = "no"
```

Comment:
You should have the value of `"secure"` match the value of `enable_secure_boot` passed to `get_libvirt_firmware_config`.